guides/micronaut-data-jdbc-repository/java/src/main/java/example/micronaut/GenreController.java

```diff
@@ -0,0 +1,84 @@
+package example.micronaut;
+
+import example.micronaut.domain.Genre;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.scheduling.TaskExecutors;
+import io.micronaut.scheduling.annotation.ExecuteOn;
+import io.micronaut.data.exceptions.DataAccessException;
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+
+@ExecuteOn(TaskExecutors.IO)  // <1>
+@Controller("/genres")  // <2>
+public class GenreController {
+
+    protected final GenreRepository genreRepository;
+
+    public GenreController(GenreRepository genreRepository) { // <3>
+        this.genreRepository = genreRepository;
+    }
+
+    @Get("/{id}") // <4>
+    public Genre show(Long id) {
+        return genreRepository
+                .findById(id)
+                .orElse(null); // <5>
+    }
```

Comment:
Isn't it better to just return `Optional<Genre>`?

Reply:
changed